File: dsmart-module-converter/dsmart-module-converter-common/src/main/kotlin/ru/datana/smart/ui/converter/common/models/SignalerModel.kt
package ru.datana.smart.ui.converter.common.models

data class SignalerModel(
    var level: SignalerLevelModel = SignalerLevelModel.NONE,
    var sound: SignalerSoundModel = SignalerSoundModel.NONE
) {
    companion object {
        val NONE = SignalerModel()
    }

    enum class SignalerLevelModel {
        NONE,
        INFO,
        WARNING,
        CRITICAL
    }
}

File: dsmart-module-converter/dsmart-module-converter-widget/projects/datana-smart/converter-widget/src/lib/models/signaler.model.ts
import {SignalerLevelModel} from "./signaler-level.model";
import {SignalerSoundModel} from "./signaler-sound.model";

export class SignalerModel {

  constructor(
    public level: SignalerLevelModel,
    public sound: SignalerSoundModel
  ) {
  }

}
